### PR TITLE
feat(saved-searches): Refactor and move pin + create saved search to store

### PR DESCRIPTION
**Pull request by billyvg**
Adds actions to (un)pin and create saved searches to store

Requires #12736

### CODE REVIEW

File: src/sentry/static/sentry/app/actionCreators/savedSearches.jsx [hunk 1]
```diff
@@ -10,14 +10,21 @@ export function fetchSavedSearches(api, orgId, useOrgSavedSearches = false) {
     data.use_org_level = '1';
   }
 
+  SavedSearchesActions.fetchSavedSearches();
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/savedSearches.jsx]: The first time I read this, I thought this did the XHR request. Would a name like `startFetchSavedSearches()` be less confusing?

File: src/sentry/static/sentry/app/actionCreators/savedSearches.jsx [hunk 2]
```diff
@@ -10,14 +10,21 @@ export function fetchSavedSearches(api, orgId, useOrgSavedSearches = false) {
...
-  promise.then(res => {
-    SavedSearchActions.updateSavedSearches(res);
-  });
+  promise
+    .then(resp => {
+      SavedSearchesActions.fetchSavedSearchesSuccess(resp);
+    })
+    .catch(err => {
+      SavedSearchesActions.fetchSavedSearchesError(err);
+      // TODO: Add error toast?
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/savedSearches.jsx]: :+1:

File: src/sentry/static/sentry/app/utils/withSavedSearches.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    render() {
+      return (
+        <Feature features={['org-saved-searches']}>
+          {({hasFeature}) => {
+            const {params} = this.props;
+            const {searchId} = params;
+            const {savedSearches, isLoading} = this.state;
+            let savedSearch = null;
+            // Switch to the the current saved search or pinned result if available
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/withSavedSearches.jsx]:
```suggestion
            // Switch to the current saved search or pinned result if available
```

File: tests/js/spec/stores/savedSearchesStore.spec.jsx [hunk 4]
```diff
@@ -1,28 +1,237 @@
...
+          type: 0,
+          query: 'assigned:me',
+        },
+        ...searches,
+      ],
+    });
+    fetchSavedSearches(api, 'org-1');
+    await tick();
+
+    unpinSearch(api, 'org-1', 0, searches[0]);
```
Comment (markstory) [inline tests/js/spec/stores/savedSearchesStore.spec.jsx]: Will our client-side state manipulation handle the scenario where a user un-pins a search that is also an organization saved search?

Comment (billyvg) [reply tests/js/spec/stores/savedSearchesStore.spec.jsx]: Yeah, there's a test for that: https://github.com/getsentry/sentry/pull/12737/files/553e943638a1dd1534d847d69b18a185d4d53d67#diff-cb2c47c4efc1613f2aacb3629fb9c899R198

File: src/sentry/static/sentry/app/stores/savedSearchesStore.jsx [hunk 5]
```diff
@@ -1,34 +1,165 @@
...
-    this.reset();
+  /**
+   * If pinned search, remove from list if `isPrivate` is true (meaning user created as pin).
+   * Otherwise change `isPinned` to false (e.g. if it's default or org saved search)
+   */
+  getFilteredSearches(type, existingSearchId) {
+    return this.state.savedSearches
+      .filter(
+        savedSearch =>
+          !(savedSearch.isPinned && savedSearch.type === type && savedSearch.isPrivate)
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/stores/savedSearchesStore.jsx]: Ah, thanks

File: tests/js/spec/stores/savedSearchesStore.spec.jsx (hunk 4 above)
Comment (billyvg) [reply tests/js/spec/stores/savedSearchesStore.spec.jsx]: Ah nm, I have my flags all mixed up.

Comment (billyvg) [reply tests/js/spec/stores/savedSearchesStore.spec.jsx]: Fixed up the flags and updated tests https://github.com/getsentry/sentry/pull/12737/commits/813730035392941ec9585ea77ffa10a6b0ba3e5f

**Review by markstory — APPROVED** [review]

**Pull request merged**